src/host/path_join.c

```diff
@@ -150,6 +150,11 @@ int path_has_deferred_join(lua_State* L)
 	return 1;
 }
 
+#ifdef __GNUC__
+/* avoid GCC warning below: 'strncpy' output truncated before terminating nul copying... */
```

Comment:
I think this warning might be highlighting a problem with our usage of `strncpy`. `strncpy(inBuffer, path, len)` is 100% wrong since `len` is based on the length of `path`, which makes it identical to `strcpy(inBuffer, path)`. What we want is to limit the copy by the length of the buffer, `strcpy(inBuffer, path, sizeof(inBuffer))`. If you change it to that, does it still emit the warning? Might need a `-1` on the size, but I don't think so.

Reply:
`strncpy` should copy the null terminator, so the only time that `inBuffer[x] = '\0'` is needed is when `path` is larger than `inBuffer`, so `inBuffer[sizeof(inBuffer) - 1] = '\0'` should always be correct.

So, I'm pretty sure the final solution would have been:
```c
strncpy(inBuffer, path, sizeof(inBuffer));
inBuffer[sizeof(inBuffer) - 1] = '\0';
```

I am concerned with potential bugs in the `strncpy_t` you've got, maybe not in the current version but if someone comes along in the future and changes it at all. However, I'm happy to merge it as-is since there is a comment there with a link to the original version of the function which should be a hint to not modify the function.

> Also C11 has `strcpy_s` which is also interesting, but doesn't provide enough guarantees as the new proposed fix.

My understanding of the `_s` functions is that they're optional and they're mostly not implemented. In addition to that, they're different to the Microsoft ones which have existed longer, and obviously, won't be replaced.